R/monthly_downscaling.R

```diff
@@ -129,3 +129,53 @@ monthly_downscaling <- function(frac, fld_data, fld_coordinates, fld_time, var){
   list(data = monthly_data, coordinates = fld_coordinates)
 
 }
+
+
+#' Wrapper function for running downscaling from cassandra component
+#'
+#' Convert inputs received from the cassandra component to the format expected
+#' by \code{\link{monthly_downscaling}} and pass the converted arguments.
+#'
+#' The conversions performed are
+#' \describe{
+#' \item{frac}{Dataset is looked up by name.}
+#' \item{fld_data}{The downscaling function wants a single matrix.  This
+#' function accepts a list of matrices and calls the downscaling function on
+#' each one sequentially.}
+#' \item{coords}{The matrix of coordinate data is converted to a data
+#' frame, and the index column is added.}
+#' \item{time}{No conversion.}
+#' \item{var}{No conversion.}
+#' }
+#'
+#' @param frac Name of monthly fraction dataset to use.
+#' @param fld_data List of matrices of annual field data to downscale to monthly.
+#' @param coords Coordinate matrix for the field data, such as the one
+#' produced by \code{\link[fldgen]{coord_array}}.
+#' @param time Vector of years for the input field matrices.
+#' @param var String indicating the variable being downscaled.  Either 'tas' or
+#' 'pr'.
+#' @return List of matrices of downscaled data.  Grid cells are in rows, months
+#' in columns (This is the opposite of the convention used everywhere else in
+#' this package.)
+#' @export
+#' @keywords internal
+downscaling_component_api <- function(frac, fld_data, coords, time, var)
+{
+    frac <- get(frac)   # would be better to use getFromNamespace, but it doesn't seem to work.
+
+    if(is.null(colnames(coords)))
+        colnames(coords) <- c('lat','lon') # Won't be necessary if you use the
+                                        # output of coord_array
+    fld_coordinates <- as.data.frame(coords)
+    fld_coordinates$column_index <- seq(nrow(fld_coordinates))
+
+    ## return the list of downscaled fields, transposed into the arrangement
+    ## needed by downstream users
+    lapply(fld_data,
+           function(f) {
+               md <- monthly_downscaling(frac, f, fld_coordinates, time, var)
+               t(md$data)
+           })
+
+}
```

Comment:
@rplzzz This looks good and addresses my concerns about what Casandra is going to interact with. Do we need to make a version of this to convert to the Xanthos units?

Reply:
Probably.  I'll take a look at where it would fit in best.